server/index.py

```diff
@@ -0,0 +1,333 @@
+        new_video_path = f'./vite/public/{recording_id}-enhanced.mp4'
+        final_clip.write_videofile(new_video_path)
+        
+        video_name = f'{recording_id}.mp4'
+        original_name = f'original-{video_name}'
+        original_path = f'./vite/public/{original_name}'
+        shutil.move(video_path, original_path)
+        
+        json_dict = json_dict | { 'processed_video': f'{recording_id}-enhanced.mp4', 'original_video': original_name}
+
+        response = json_dict
+        return jsonify(response), 200
+    except Exception as e:
+        return process_error('failed to process file – check logs for details', e)
+
+def process_video_dubbing(recording_id: str, video_path: str) -> tuple[quart.Response, int]:
+    """Runs filler-word-removal processing on given file."""
+    try:
+        video_name = f'{recording_id}.mp4'
+        clip = VideoFileClip(video_path)
+        duration = clip.duration
+        clip = clip.subclip(1,duration)
+        original_name = f'original-{video_name}'
+        original_trimmed_video = f'./vite/public/{original_name}'
+        clip.write_videofile(original_trimmed_video)
+
+        video = sieve.Video(path=original_trimmed_video)
+        path = video_dubbing(video, "spanish")
+        processed_name = f'dubbed-{video_name}'
+        processed_path = f'./vite/public/{processed_name}'
+        shutil.move(path, processed_path)
+        json_dict = {
+            recording_id: recording_id,
+            'original_video': original_name,
+            'processed_video': processed_name
+        }
+        print()
+
+        response = json_dict
+        return jsonify(response), 200
+    except Exception as e:
+        return process_error('failed to process file – check logs for details', e)
+    
+def process_transcript_analyzer(recording_id: str, video_path: str) -> tuple[quart.Response, int]:
+    """Runs filler-word-removal processing on given file."""
+    try:
+        video = sieve.Video(path=video_path)
+        video_transcript_analyzer = sieve.function.get("sieve/video_transcript_analyzer")
+        output = video_transcript_analyzer.run(video)
+        json_dict = {
+            recording_id: recording_id
+        }
+        json_path = os.path.join(get_upload_dir_path(), f'{recording_id}.json')
+
+        for i, output_object in enumerate(output):
+            print(i)
+            print(output_object)
+            print(type(output_object))
+            if i == 2 or i == 3 or i == 4:
+                json_dict = json_dict | output_object
+
+        with open(json_path, 'w') as fp:
+            json.dump(json_dict, fp)
+
+            response = json_dict
+            return jsonify(response), 200
+    except Exception as e:
+        return process_error('failed to process file – check logs for details', e)
+
+def transcript_to_text(transcript):
+    text = " ".join([segment["text"] for segment in transcript])
+    return text
+
+@sieve.function(
+    name="video-dubber",
```

Comment:
We've used a few sieve functions up until this point, but this is the first time I see this decorator and it is for use one func. Wondering what it is used for and why just this one?  Worth adding a comment imo

Reply:
This is taken from Sieve's documentation, TBH I'm not sure why they wrote it this way?

Reply:
Hmm ok; I'm a little worried that this does not seem to be version locked at all and the demo could break as soon as that function changes?